rs/bitcoin/ckbtc/minter/src/updates/retrieve_btc.rs

```diff
@@ -299,6 +299,31 @@ pub async fn retrieve_btc_with_approval(
         ));
     }
 
+    let new_kyt_principal = read_state(|s| {
+        s.new_kyt_principal
+            .expect("BUG: upgrade procedure must ensure that the new KYT principal is set")
+            .get()
+            .into()
+    });
+
+    match new_kyt_check_address(new_kyt_principal, args.address.clone()).await {
```

Comment:
Here for a lack of a better place: Shouldn't we remove the block L.266, since this should be taken care of by the new KYT, no?
```rust
 if crate::blocklist::BTC_ADDRESS_BLOCKLIST
        .binary_search(&args.address.trim())
        .is_ok()
    {
        ic_cdk::trap("attempted to retrieve BTC to a blocked address");
    }
```

Reply:
nit: I think it would be nicer if `new_kyt_check_address` would use as parameter the `parsed_address` from L.294 instead of just a String. It's usually safer to passer around instances that have been validated instead of just `String`.

Reply:
> Shouldn't we remove the block L.266

Indeed, the block list can be removed from the minter crate. Thanks for catching it!

> I think it would be nicer if new_kyt_check_address would use as parameter the parsed_address from L.294

Good point! Fixed.